File: adventjs/2025/20/solution.ts
function dropGifts(warehouse: string[][], drops: number[]): string[][] {
  const result = warehouse.map(row => [...row]);
  
  for (const col of drops) {
    for (let row = result.length - 1; row >= 0; row--) {
      if (result[row][col] === '.') {
        result[row][col] = '#';
        break;
      }
    }
  }
  
  return result;
}

File: adventjs/2025/20/solution.py
def drop_gifts(warehouse: list[list[str]], drops: list[int]) -> list[list[str]]:
  result = [row[:] for row in warehouse]
  
  for col in drops:
    for row in range(len(result) - 1, -1, -1):
      if result[row][col] == '.':
        result[row][col] = '#'
        break
  
  return result
